src/main/java/appeng/me/storage/BasicCellInventory.java

```diff
@@ -165,11 +188,11 @@ public IAEItemStack injectItems( final IAEItemStack input, final Actionable mode
 			{
 				if( input.getStackSize() > remainingItemCount )
 				{
-					final IAEItemStack toReturn = input.copy();
+					final T toReturn = input.copy();
 					toReturn.setStackSize( input.getStackSize() - remainingItemCount );
 					if( mode == Actionable.MODULATE )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)